File: app/src/main/java/org/stepik/android/adaptive/pdd/data/model/Attempt.java
package org.stepik.android.adaptive.pdd.data.model;

public final class Attempt {
    private long id;
    private long step;
    private Dataset dataset;
    private String dataset_url;
    private String time;
    private String status;
    private String time_left;
    private long user;

    public Attempt(final long step) {
        this.step = step;
    }

    public long getStep() {
        return step;
    }

    public Dataset getDataset() {
        return dataset;
    }

    public long getId() {
        return id;
    }
}

File: app/src/main/java/org/stepik/android/adaptive/data/model/Attempt.kt
package org.stepik.android.adaptive.data.model

import com.google.gson.annotations.SerializedName

class Attempt (
        val id: Long = 0,
        val step: Long = 0,
        @SerializedName("dataset")
        val datasetWrapper: DatasetWrapper? = null,
        val dataset_url: String? = null,
        val time: String? = null,
        val status: String? = null,
        val time_left: String? = null,
        val user: Long = 0
) {
    fun getDataset() = datasetWrapper?.dataset
}
